tests/lax_numpy_test.py

```diff
@@ -2299,6 +2299,35 @@ def testHistogram(self, shape, dtype, bins, density, weights):
                               tol=tol)
     self._CompileAndCheck(jnp_fun, args_maker)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": "_{}_bins={}_weights={}_density={}".format(
+      jtu.format_shape_dtype_string(shape, dtype), bins, weights, density),
+      "shape": shape,
+      "dtype": dtype,
+      "bins": bins,
+      "weights": weights,
+      "density": density
+    }
+    for shape in [(5, 3), (10, 3)]
+    for dtype in int_dtypes
+    # We only test explicit integer-valued bin edges because in other cases
+    # rounding errors lead to flaky tests.
+    for bins in [(2, 2, 2), [[-5, 0, 4], [-4, -1, 2], [-6, -1, 4]]]
+    for weights in [False, True]
+    for density in [False]
+  ))
+  def testHistogramdd(self, shape, dtype, bins, weights, density):
+    rng = jtu.rand_default(self.rng())
+    _weights = lambda w: abs(w) if weights else None
+    np_fun = lambda a, w: np.histogramdd(a, bins=bins, weights=_weights(w), density=density)
+    jnp_fun = lambda a, w: jnp.histogramdd(a, bins=bins, weights=_weights(w), density=density)
+    args_maker = lambda: [rng(shape, dtype), rng((shape[0],), dtype)]
+    tol = {jnp.bfloat16: 2E-2, np.float16: 1E-1}
+    self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker, check_dtypes=False,
+                              tol=tol)
+    # histogramdd uses ravel_multi_index with mode = 'raise' so cannot perform this check
```

Comment:
It might be worth using a different mode to allow compilation. Would `mode='clip'` be appropriate? Do we ever expect out-of-bounds indices within the `histogramdd` computation?

Reply:
I'll give it a try

Reply:
I tried making the change   
`xy = ravel_multi_index(bin_idx_by_dim, nbins, mode='clip')`
and uncommenting the `_CompileAndCheck` test but end up with that JIT error 
```

jax.traceback_util.FilteredStackTrace: jax.core.ConcretizationTypeError: Abstract tracer value encountered where concrete value is expected.

The problem arose with the `bool` function. 
```

Reply:
It would be worth figuring out what's causing this, and trying to fix it. `jnp.histogram` is compatible with JIT, so I don't think there should be any reason that `jnp.histogramdd` could not be.